controlplane/pkg/nsmd/nsmd.go

```diff
@@ -574,3 +580,29 @@ func (nsm *nsmServer) StartAPIServerAt(ctx context.Context, sock net.Listener, p
 	}()
 	span.Logger().Infof("NSM gRPC API Server: %s is operational", sock.Addr().String())
 }
+
+// WorkspaceFromContext finds workspace by clientToken stored in the Context
+func (nsm *nsmServer) WorkspaceFromContext(ctx context.Context) *Workspace {
+	if clientToken := tools.ClientToken(ctx); len(clientToken) > 0 {
+		for _, w := range nsm.workspaces {
+			if w.clientToken == clientToken {
+				return w
+			}
+		}
+	}
+	return nil
+}
+
+// WorkspaceNameByGRPCContext finds workspace by its client token passed via GRPC context
+func (nsm *nsmServer) WorkspaceNameByGRPCContext(ctx context.Context) string {
+	if clientToken := tools.ClientToken(ctx); len(clientToken) > 0 {
```

Comment:
Why don't you use `WorkspaceFromContext` in that method?

Reply:
hm.. good point, thanks )

- [ ] use WorkspaceFromContext